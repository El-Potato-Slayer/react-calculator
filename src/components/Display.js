import PropTypes from 'prop-types';

function Display({ data }) {
  let { total, next, operation } = data;
  if (!total) {
    total = '0';
  }
  if (!operation) {
    operation = '';
  }
  if (!next) {
    next = '';
  }
  return (

    <div>
      {
        total
      }
      {
        operation
      }
      {
        next
      }
    </div>
  );
}

Display.propTypes = {
  data: PropTypes.string,
};

Display.defaultProps = {
  data: '0',
};

export default Display;
