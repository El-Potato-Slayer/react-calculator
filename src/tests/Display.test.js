import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import Display from '../components/Display';

afterEach(cleanup);

it('Renders Display component without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Display data={{ total: '1', next: '5', operation: '+' }} />, div);
});

it('ensures total is 1', () => {
  const { getByTestId } = render(<Display data={{ total: '1', next: '5', operation: '+' }} />);
  expect(getByTestId('display')).toHaveTextContent('1');
});

it('ensures total is not 2', () => {
  const { getByTestId } = render(<Display data={{ total: '1', next: '5', operation: '+' }} />);
  expect(getByTestId('display')).not.toHaveTextContent('2');
});

it('ensures next is 5', () => {
  const { getByTestId } = render(<Display data={{ total: '1', next: '5', operation: '+' }} />);
  expect(getByTestId('display')).toHaveTextContent('5');
});

it('ensures next is not 4', () => {
  const { getByTestId } = render(<Display data={{ total: '1', next: '5', operation: '+' }} />);
  expect(getByTestId('display')).not.toHaveTextContent('4');
});

it('ensures operation is not -', () => {
  const { getByTestId } = render(<Display data={{ total: '1', next: '5', operation: '+' }} />);
  expect(getByTestId('display')).not.toHaveTextContent('-');
});

it('matches snapshot', () => {
  const tree = renderer.create(<Display data={{ total: '1', next: '5', operation: '+' }} />).toJSON();
  expect(tree).toMatchSnapshot();
});
