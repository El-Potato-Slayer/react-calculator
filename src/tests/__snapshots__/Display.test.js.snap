// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="display"
  data-testid="display"
>
  <span>
    1
    +
    5
  </span>
</div>
`;
